lib/pure/asynchttpserver.nim

```diff
@@ -71,6 +72,20 @@ type
     maxBody: int ## The maximum content-length that will be read for the body.
     maxFDs: int
 
+func getSocket*(a: AsyncHttpServer): AsyncSocket {.since: (1, 5, 1).} =
+  ## Returns the ``AsyncHttpServer``s internal ``AsyncSocket`` instance.
+  ## 
+  ## Useful for identifying what port the AsyncHttpServer is bound to, if it
+  ## was chosen automatically.
+  runnableExamples:
+    import asyncdispatch # Port
+    import asyncnet # getLocalAddr
+    import nativesockets # AF_INET
+    let server = newAsyncHttpServer()
+    server.listen(Port(0)) # Socket is not bound until this point
+    let port = getLocalAddr(server.getSocket.getFd, AF_INET)[1]
```

Comment:
use self documenting code instead of comments when possible:
```suggestion
    from asyncdispatch import Port
    from asyncnet import getLocalAddr
    from nativesockets import AF_INET
    let server = newAsyncHttpServer()
    server.listen(Port(0)) # Socket is not bound until this point
    let port = getLocalAddr(server.getSocket.getFd, AF_INET)[1]
    doAssert port > 0
    server.close()
```
(I also added server.close(), please double check if correct)

hopefully we can follwup quickly with a followup PR to make this easier for client code, PR welcome

Reply:
Fixed.